src/main/java/uk/gov/hmcts/reform/pip/account/management/validation/validator/ValueOfEnumValidator.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.pip.account.management.validation.validator;
+
+import uk.gov.hmcts.reform.pip.account.management.validation.annotations.ValueOfEnum;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * Validator class that checks if an input is in the supplied enum.
+ */
+public class ValueOfEnumValidator implements ConstraintValidator<ValueOfEnum, CharSequence> {
+    private List<String> acceptedValues;
+
+    @Override
+    public void initialize(ValueOfEnum annotation) {
+        acceptedValues = Stream.of(annotation.enumClass().getEnumConstants())
+            .map(Enum::name)
+            .collect(Collectors.toList());
```

Comment:
Now we have moved to Java 17, sonarcloud will likely to log a code smell when we use "collect(Collectors.to List())". Can just use "toList()"

Reply:
Ah yes, done :)